src/tests.c

```diff
@@ -4324,7 +4324,7 @@ void test_ecdsa_sign_verify(void) {
     secp256k1_scalar one;
     secp256k1_scalar msg, key;
     secp256k1_scalar sigr, sigs;
-    int recid;
+    int recid = -1;
     int getrec;
     random_scalar_order_test(&msg);
```

Comment:
@gmaxwell Is this your suggestion?
```suggestion
    int getrec;
    /* Initialize recid to suppress a false positive -Wconditional-uninitialized in clang.
       VG_UNDEF ensures that valgrind will still treat the variable as uninitialized. */
    int recid = -1; VG_UNDEF(&recid, sizeof(recid));
    random_scalar_order_test(&msg);
```

Reply:
yep!

Reply:
Thank you. I amended the first commit with your changes.